solc/CommandLineParser.cpp

```diff
@@ -588,6 +592,17 @@ General Information)").c_str(),
 			"Select desired EVM version. Either homestead, tangerineWhistle, spuriousDragon, "
 			"byzantium, constantinople, petersburg, istanbul, berlin, london or paris."
 		)
+	;
+	if (!_forHelp) // Note: We intentionally keep this undocumented for now.
+		outputOptions.add_options()
+			(
+				g_strEOFVersion.c_str(),
+				po::value<string>()->value_name("version"),
```

Comment:
You could give it an implicit value. This way, if user specifies `--eof-version` without a number, it defaults to 1 but remains unset if the option is not specified.
```suggestion
				g_strEOFVersion.c_str(),
				po::value<string>()->value_name("version")->implicit_value(1),
```